src/node/interfaces.cpp

```diff
@@ -536,6 +538,20 @@ class ChainImpl : public Chain
         }
         return std::nullopt;
     }
+    bool hasBlockFilterIndex(BlockFilterType filter_type) override
+    {
+        return GetBlockFilterIndex(filter_type) != nullptr;
+    }
+    std::optional<bool> blockFilterMatchesAny(BlockFilterType filter_type, const uint256& block_hash, const GCSFilter::ElementSet& filter_set) override
+    {
+        const BlockFilterIndex* block_filter_index{GetBlockFilterIndex(filter_type)};
+        if (!block_filter_index) return std::nullopt;
+
+        BlockFilter filter;
+        const CBlockIndex* index{WITH_LOCK(::cs_main, return chainman().m_blockman.LookupBlockIndex(block_hash))};
+        if (!block_filter_index->LookupFilter(index, filter)) return std::nullopt;
```

Comment:
As `LookupBlockIndex` can return a nullptr, better if we check it. (Currently, it cannot fail due our single use case but it might fail in the future)